Demo/src/main/java/com/braintreepayments/demo/GooglePayFragment.java

```diff
@@ -38,11 +42,13 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         googlePayClient = new GooglePayClient(requireContext(), super.getAuthStringArg());
         googlePayLauncher = new GooglePayLauncher(this,
                 paymentAuthResult -> googlePayClient.tokenize(paymentAuthResult,
-                        (paymentMethodNonce, error) -> {
-                            if (error != null) {
-                                handleError(error);
-                            } else {
-                                handleGooglePayActivityResult(paymentMethodNonce);
+                        (googlePayResult) -> {
+                            if (googlePayResult instanceof GooglePayResult.Failure) {
+                                handleError(((GooglePayResult.Failure) googlePayResult).getError());
+                            } else if (googlePayResult instanceof GooglePayResult.Success){
+                                handleGooglePayActivityResult(((GooglePayResult.Success) googlePayResult).getNonce());
+                            } else if (googlePayResult instanceof GooglePayResult.Cancel) {
```

Comment:
should we add an `else` block for other values, or are these the only options from google?

Reply:
We define the `GooglePayResult` object as a sealed class so these are the only options.